### PR TITLE
Enabled .NET remote host by default

### PR DESCRIPTION
**Pull request by genlu**
Finally able to do this, now that we have an arm64 vsix for remote bits. Also changed the option from a feature flag to a regular VS option. We might want run on .NET exclusively at some point in the future, but for now I'm keeping the option.

Validation:
~~https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/471195~~
https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/471960

UI change in options panel:

image

RPS exception tracking bugs:
[AB#1820959](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1820959)
[AB#1820967](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1820967)

### CODE REVIEW

**Comment by genlu** [comment]
@MartyIX That is correct.

File: src/VisualStudio/Core/Def/ServicesVSResources.resx [hunk 1]
```diff
@@ -1813,6 +1813,9 @@ Additional information: {1}</value>
   <data name="Run_code_analysis_in_separate_process_requires_restart" xml:space="preserve">
     <value>Run code analysis in separate process (requires restart)</value>
   </data>
+  <data name="Run_code_analysis_on_dotnet_requires_restart" xml:space="preserve">
+    <value>Run code analysis on .NET (requires restart)</value>
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/ServicesVSResources.resx:1817]: I can get the desire to not say ".NET Core", but this text might be a bit of a head scratcher for folks if they think unchecking this means we're running it on, uh, C++? Should we just say ".NET 6" or something so it's a bit clearer that way?

Comment (genlu) [reply src/VisualStudio/Core/Def/ServicesVSResources.resx:1817]: How about " latest .NET"? Given that we might be moving to .NET 7 soon

Comment (genlu) [reply src/VisualStudio/Core/Def/ServicesVSResources.resx]: image

Comment (jasonmalinowski) [reply src/VisualStudio/Core/Def/ServicesVSResources.resx]: @genlu Oh I like that!

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs [hunk 2]
```diff
@@ -325,5 +326,15 @@ private void EnterOutliningMode_Unchecked(object sender, RoutedEventArgs e)
...
+
+        private void Run_code_analysis_in_separate_process_Checked(object sender, RoutedEventArgs e)
+        {
+            Run_code_analysis_on_dotnet.IsEnabled = true;
+        }
+
+        private void Run_code_analysis_in_separate_process_Unchecked(object sender, RoutedEventArgs e)
+        {
+            Run_code_analysis_on_dotnet.IsEnabled = false;
+        }
```
Comment (sharwell) [inline src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs:338]: 📝 I'd prefer to not do this

File: src/VisualStudio/VisualBasic/Impl/Options/AdvancedOptionPageControl.xaml.vb [hunk 3]
```diff
@@ -237,5 +239,13 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Options
...
             Collapse_metadata_signature_files_on_open.IsEnabled = False
         End Sub
+
+        Private Sub Run_code_analysis_in_separate_process_Checked(sender As Object, e As RoutedEventArgs)
+            Run_code_analysis_on_dotnet.IsEnabled = True
+        End Sub
+
+        Private Sub Run_code_analysis_in_separate_process_Unchecked(sender As Object, e As RoutedEventArgs)
+            Run_code_analysis_on_dotnet.IsEnabled = False
+        End Sub
```
Comment (sharwell) [inline src/VisualStudio/VisualBasic/Impl/Options/AdvancedOptionPageControl.xaml.vb:249]: 📝 Also would like these removed

File: src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs (hunk 2 above)
Comment (genlu) [reply src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs:338]: I added it just in case. We can always remove it in before GA

**Pull request merged**